util/git/client.go

```diff
@@ -108,21 +107,14 @@ type runOpts struct {
 }
 
 var (
-	maxAttemptsCount = 1
+	maxAttemptsCount int64
 	maxRetryDuration time.Duration
 	retryDuration    time.Duration
 	factor           int64
 )
 
 func init() {
-	if countStr := os.Getenv(common.EnvGitAttemptsCount); countStr != "" {
-		if cnt, err := strconv.Atoi(countStr); err != nil {
-			panic(fmt.Sprintf("Invalid value in %s env variable: %v", common.EnvGitAttemptsCount, err))
-		} else {
-			maxAttemptsCount = int(math.Max(float64(cnt), 1))
-		}
-	}
-
+	maxAttemptsCount = env.ParseInt64FromEnv(common.EnvGitAttemptsCount, 1, 1, math.MaxInt64)
```

Comment:
Any particular reason that you've not used `env.ParseNumFromEnv()` instead? That would give you an `int` as a return value natively, and you wouldn't have to cast later on.

The current conversion is potentially bound to fail, because you allow the maximum value as `math.MaxInt64` but cast to an `int` further down in the code.